File: javascript/prime-factors/prime-factors.js
var PrimeFactors = function() {};

PrimeFactors.for = function(num) {
    solution = [];
    var i = 2;
    while (num > 1) {
        if (num % i === 0) {
            solution.push(i)
            num = num / i;
            i = 1;
        }
        i++
    }
    return solution
}

module.exports = PrimeFactors;

File: ruby/prime-factors/prime_factors.rb
class PrimeFactors
  def self.for(num)
    prime_factors = []
    return prime_factors if num < 2

    while num > 1
      n = (2..num).find { |divisor| (num % divisor).zero? }
      prime_factors << n
      num /= n
    end
    prime_factors
  end
end
